### PR TITLE
add gergo:mim layout

### PR DESCRIPTION
**Pull request by Marthin-**
## Description

Added a new keymap for gergo keyboard, focused on us-intl layout and a few sysadmin shortcuts
## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [ ] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/gergo/keymaps/mim/config.h [hunk 1]
```diff
@@ -0,0 +1,9 @@
+#pragma once
```
Comment (drashna) [inline keyboards/gergo/keymaps/mim/config.h:1]: Could you add a GPL2+ compatible license header here? 

For instance: 
https://github.com/qmk/qmk_firmware/blob/28ff51175b6d45ce5a8dbfe7ed6e7a12df3bc8a8/quantum/template/base/keyboard.c#L1-L15

File: keyboards/gergo/keymaps/mim/config.h [hunk 2]
```diff
@@ -0,0 +1,9 @@
...
+
+//#define IGNORE_MOD_TAP_INTERRUPT
+#define COMBO_TERM 125
+#define COMBO_COUNT 3
+#define PERMISSIVE_HOLD
+#define DEBOUNCING_DELAY 10
```
Comment (drashna) [inline keyboards/gergo/keymaps/mim/config.h:7]:
```suggestion
#define DEBOUNCE 10
```

File: keyboards/gergo/keymaps/mim/keymap.c [hunk 3]
```diff
@@ -0,0 +1,48 @@
+#include QMK_KEYBOARD_H
```
Comment (drashna) [inline keyboards/gergo/keymaps/mim/keymap.c:1]: Could you add a GPL2+ compatible license header here? 

For instance: 
https://github.com/qmk/qmk_firmware/blob/28ff51175b6d45ce5a8dbfe7ed6e7a12df3bc8a8/quantum/template/base/keyboard.c#L1-L15

File: keyboards/gergo/keymaps/mim/keymap.c [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+    TMUX = SAFE_RANGE,
+};
+
+bool process_record_user(uint16_t keycode, keyrecord_t *record) {
+    switch (keycode) {
+    case TMUX:
+        if (record->event.pressed) {
+            SEND_STRING(SS_LCTL("x"));
+        } else {
+        }
```
Comment (drashna) [inline keyboards/gergo/keymaps/mim/keymap.c:29]:
```suggestion
        }
```

File: keyboards/gergo/keymaps/mim/keymap.c [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+        break;
+    } return true; };
```
Comment (drashna) [inline keyboards/gergo/keymaps/mim/keymap.c:31]:
```suggestion
    } 
    return true;
}; 
```

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/gergo/keymaps/mim/config.h (hunk 1 above)
Comment (Marthin-) [reply keyboards/gergo/keymaps/mim/config.h:17]: Done !
Thanks for reviewing :)

Comment (walli55) [reply keyboards/gergo/keymaps/mim/config.h:17]: Alomaisi

**Comment by Marthin-** [comment]
@drashna it should be okay now :)

File: keyboards/gergo/keymaps/mim/rules.mk [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+VERBOSE 		 = yes
+DEBUG_MATRIX_SCAN_RATE   = no
+DEBUG_BALLER 		 = no
+DEBUG_MATRIX		 = yes
+
+# A bunch of stuff that you shouldn't touch unless you
+# know what you're doing.
+#
+# No touchy, capiche?
+SRC += matrix.c i2c_master.c
```
Comment (drashna) [inline keyboards/gergo/keymaps/mim/rules.mk:29]: This should be removed, as it exists at the kb level, so no need to re-include it here. 
```suggestion
```

**Comment by Marthin-** [comment]
@drashna could you merge this when you have time ? :) Thanks a lot

**Review by drashna — APPROVED** [review]

**Comment by Marthin-** [comment]
Could someone merge this since it's been approved ? :)
Thanks in advance

File: keyboards/gergo/keymaps/mim/README.md [hunk 7]
```diff
@@ -0,0 +1,85 @@
+# Magui's International Map
```
Comment (fauxpark) [inline keyboards/gergo/keymaps/mim/README.md:1]: Please rename this file to lowercase.

File: keyboards/gergo/keymaps/mim/keymap.c [hunk 8]
```diff
@@ -0,0 +1,64 @@
...
+
+enum custom_keycodes {
+    TMUX = SAFE_RANGE,
+};
+
+bool process_record_user(uint16_t keycode, keyrecord_t *record) {
+    switch (keycode) {
+    case TMUX:
+        if (record->event.pressed) {
+            SEND_STRING(SS_LCTL("x"));
```
Comment (fauxpark) [inline keyboards/gergo/keymaps/mim/keymap.c:43]: Why not just put `C(KC_X)` directly in the keymap?

Comment (Marthin-) [reply keyboards/gergo/keymaps/mim/keymap.c]: I recycled this snippet for longer SEND_STRING that isn't used anymore. fixed.

File: keyboards/gergo/keymaps/mim/README.md [hunk 9]
```diff
@@ -0,0 +1,85 @@
+# magui's international map
```
Comment (fauxpark) [inline keyboards/gergo/keymaps/mim/README.md:1]: Could you rename this file to lowercase?

**Review by fauxpark — CHANGES_REQUESTED** [review]

**Pull request closed without merging**